### ISSUE TITLE
Login no longer processing as of 16 May 2018

### ISSUE THREAD

**Issue by drmstwizard** [yakyak/yakyak#918]
The Windows (32) desktop client is no longer loading/logging in. I tried reinstalling, and the same problem comes up. I've been using the client with no problems since 2015.

**Comment by chazzzle** [comment]
same here

**Comment by Midda-C** [comment]
Can confirm.  Just stuck on "Connecting" here.

**Comment by colinmo** [comment]
Confirming the same happening with the OS X YakYak client.

**Comment by r0bnewt0n** [comment]
Same issue on Linux client

**Comment by blackroseMD1** [comment]
Same issue here on Windows 10 x64

**Comment by hypercube33** [comment]
Happening on Windows 10 x64 with 1.4.4 and 1.5.0 even after clearing appdata. Neato.

**Comment by Aurezalia** [comment]
Not working on Ubuntu 16.04 or Windows 10 x64 here, either! Just stuck on "connecting".

**Comment by Viper007Bond** [comment]
Obviously the problem is widespread and not OS-specific, so rather than spamming "me too" comments, let's just hit that 👍 icon instead. 😉

**Comment by aurelijusrozenas** [comment]
I guess same here: Ubuntu 17.10
```
connect_failed SyntaxError: Unexpected token <
    at Object.bind.out (/snap/yakyak/1/resources/app/node_modules/hangupsjs/lib/init.js:62:23)
    at /snap/yakyak/1/resources/app/node_modules/hangupsjs/lib/init.js:63:26
    at _fulfilled (/snap/yakyak/1/resources/app/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/snap/yakyak/1/resources/app/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/snap/yakyak/1/resources/app/node_modules/q/q.js:816:13)
    at /snap/yakyak/1/resources/app/node_modules/q/q.js:877:14
    at runSingle (/snap/yakyak/1/resources/app/node_modules/q/q.js:137:13)
    at flush (/snap/yakyak/1/resources/app/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
error connecting SyntaxError: Unexpected token <
    at Object.bind.out (/snap/yakyak/1/resources/app/node_modules/hangupsjs/lib/init.js:62:23)
    at /snap/yakyak/1/resources/app/node_modules/hangupsjs/lib/init.js:63:26
    at _fulfilled (/snap/yakyak/1/resources/app/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/snap/yakyak/1/resources/app/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/snap/yakyak/1/resources/app/node_modules/q/q.js:816:13)
    at /snap/yakyak/1/resources/app/node_modules/q/q.js:877:14
    at runSingle (/snap/yakyak/1/resources/app/node_modules/q/q.js:137:13)
    at flush (/snap/yakyak/1/resources/app/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by bogdan-radocea** [comment]
Same here, on ubuntu 17.10.
Also same error when starting from console:
connect_failed SyntaxError: Unexpected token <
    at Object.bind.out (/opt/yakyak/resources/app/node_modules/hangupsjs/lib/init.js:62:23)
    at /opt/yakyak/resources/app/node_modules/hangupsjs/lib/init.js:63:26
    at _fulfilled (/opt/yakyak/resources/app/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/opt/yakyak/resources/app/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/opt/yakyak/resources/app/node_modules/q/q.js:816:13)
    at /opt/yakyak/resources/app/node_modules/q/q.js:877:14
    at runSingle (/opt/yakyak/resources/app/node_modules/q/q.js:137:13)
    at flush (/opt/yakyak/resources/app/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
error connecting SyntaxError: Unexpected token <
    at Object.bind.out (/opt/yakyak/resources/app/node_modules/hangupsjs/lib/init.js:62:23)
    at /opt/yakyak/resources/app/node_modules/hangupsjs/lib/init.js:63:26
    at _fulfilled (/opt/yakyak/resources/app/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/opt/yakyak/resources/app/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/opt/yakyak/resources/app/node_modules/q/q.js:816:13)
    at /opt/yakyak/resources/app/node_modules/q/q.js:877:14
    at runSingle (/opt/yakyak/resources/app/node_modules/q/q.js:137:13)
    at flush (/opt/yakyak/resources/app/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)

**Comment by mikamikem** [comment]
It appears Google added nonce to their script tags and it's breaking the regex in hangups.  If you build from source, I changed line 59 in app/node_modules/hangupsjs/lib/init.js from:

        html = html.replace(/<\/?script>/gm, '');

To:

        html = html.replace(/<\/?script( nonce="(.*?)")?>/gm, '');

I haven't written much node.js code before and I know almost nothing of coffeescript, but it appeared that the change was necessary in hangupsjs and I just submitted a pull request to do the lazy thing that I think will work:

https://github.com/yakyak/hangupsjs/pull/90

Feel free to fix that up and/or properly handle the nonce since it seems important (but currently works for me without it).

Hope that's helpful!
Michael

**Comment by Green-Star** [comment]
For Linux users who installed Yakyak from aur and don't want to rebuild from source, you can also edit the `/usr/share/yakyak/app/node_modules/hangupsjs/lib/init.js` file with @mikamikem's fix to get yakyak up again.

Hope that helps

**Comment by mbinkhorst** [comment]
For Mac users who don't want to rebuild from source, you can right-click on the yakyak.app and 'show package contents', then navigate to 'contents/resources/app/node_modules/hangupsjs/lib/init.js' and edit that file with @mikamikem's fix to get yakyak up again.

**Comment by teicher** [comment]
fine, but will you be able to publish a maintenance release on short notice ?

**Comment by thunderbong** [comment]
Thanks @bogdan-radocea. That worked for me too.

On Windows 10, the path is the same
`<yakyak folder>\resources\app\node_modules\hangupsjs\lib\init.js`

**Comment by Stanko** [comment]
Same for MacOS you can apply [fix](https://github.com/yakyak/yakyak/issues/918#issuecomment-389772323) proposed by @bogdan-radocea, by editing

```
/Applications/YakYak.app/Contents/Resources/app/node_modules/hangupsjs/lib/init.js
```

**Comment by wowthatisrandom** [comment]
worked for me on MacOS as well, thanks @mikamikem

**Comment by ascavalcante80** [comment]
worked great for me on Ubuntu 17.04. thanks @mikamikem

**Comment by johndskinner** [comment]
Any word on a maintenance release?

**Comment by zachlandes** [comment]
Same issue on macOS Sierra (10.2). Fixed it by following instructions above.

**Comment by gyde** [comment]
Debian was the same, fixed by following above instructions. Did anybody create merge request with the fix? Doesn't look like it, will do that :)

**Comment by kcolton** [comment]
@gyde Which one?

**Comment by gyde** [comment]
See https://github.com/yakyak/hangupsjs/pull/90 @kcolton

**Comment by Zloduh** [comment]
anyone knows is snap version fixed? i don't know why but it worked better then .deb one im using now.

**Comment by kwrobel** [comment]
Using Mac OS High Sierra 10.13.4 and YakYak 1.5.0 pre-release and applied the same fix from Mikamikem and it works.

**Comment by predatorian3** [comment]
Confirmed that https://github.com/yakyak/yakyak/issues/918#issuecomment-389772323 works.

**Comment by davibe** [comment]
fixed in 1.5.0 release

**Comment by teicher** [comment]
not true, 1.5.0, release in January, need manual patching.
maybe this is a typo in the fix version number?

**Comment by davibe** [comment]
1.5.0 went out of pre-re-ease 2 days ago. it's not the same built it was on january.

**Comment by teicher** [comment]
To avoid confusion, could you bump the version number ?
My old install here sais 1.5.0 in the About dialog...

**Comment by hufon** [comment]
Please name this 1.5.1 and release package.
Thks

**Comment by davibe** [comment]
I have released 1.5.1

Next time i will try to make it more clear when it's a dev/pre release or not.
